File: 코딩 기초 트레이닝/수열과 구간 쿼리2-p181923 /수열과 구간 쿼리2 .js
// https://school.programmers.co.kr/learn/courses/30/lessons/181923

function solution(arr, queries) {
    var answer = [];

    queries.forEach((q) => {
        let s = q[0];
        let e = q[1];
        let k = q[2];

        let tmp = arr.slice(s, e+1);
        tmp = tmp.filter(t => t > k);
        tmp.sort(function(a,b) {
            return Number(a) - Number(b);
        });

        if(tmp.length > 0) answer.push(tmp[0]);
        else answer.push(-1);
    });

    return answer;
}

File: 코딩 기초 트레이닝/수열과 구간 쿼리2-p181923 /수열과 구간 쿼리2.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181923

def solution(arr, queries):
    answer = []

    for index, q in enumerate(queries):
        s = q[0]
        e = q[1]
        k = q[2]

        tmp = arr[s:e+1]
        tmp = list(filter(lambda x: x > k, tmp))
        tmp.sort()

        if len(tmp) == 0:
            answer.append(-1)
        else:
            answer.append(tmp[0])

    return answer
